store/sqlstore/bot_store.go

```diff
@@ -205,7 +205,7 @@ func (us SqlBotStore) Update(bot *model.Bot) (*model.Bot, error) {
 
 	if count, err := us.GetMaster().Update(botFromModel(bot)); err != nil {
 		return nil, errors.Wrapf(err, "update: user_id=%s", bot.UserId)
-	} else if count != 1 {
+	} else if count > 1 {
```

Comment:
Maybe it's an obvious thing but, why we are not considering `count = 0` as an error?

Reply:
That's what I outlined in PR description. In a lot of cases, we are only changing the `UpdateAt` timestamp. And if 2 queries happen to run within the same millisecond, they will fail. So `count = 0` is not exactly an error, because nothing really has changed.

Reply:
I see, thanks.

Reply:
A few questions:

How are two queries coming in on the same row within such a small time frame?

If the intention was to change something, but nothing changed, isn't that an error? I haven't looked in detail, but isn't it conceivable that a single request comes in but does not result in an updated row? I imagine this can only happen through an error, which is handled above.

Reply:
> How are two queries coming in on the same row within such a small time frame?

During the test, it happens if the first update finishes in less than a millisecond. But in real world, it can always happen. The same user can have multiple sessions open, or there can be API requests.

> If the intention was to change something, but nothing changed, isn't that an error?

It depends on what is changed from the user level. The API will always pass even if you don't change any field, and that's because the `UpdateAt` field would always be updated with the current timestamp from inside the method. So it is by design that 2 concurrent updates within the same milisecond will fail, but it really shouldn't. It has been causing test failures from a very long time, and just hasn't occurred in production yet. (Or maybe it has but is a very rare event)

I requested your review because you had already reviewed the exact same PR here https://github.com/mattermost/mattermost-server/commit/3f46cf6f60863a356c8994f1df306464feb8ae25 but directly approved it then :) So actually, this was already changed once. I am just making it consistent now throughout the codebase.

Reply:
Ah, ok, got it. Makes sense. 

Not sure why these questions didn't occur to me then. I guess they did but I figured them out myself and then promptly forgot. I'm slow on Mondays ;)